### DISCUSSION TITLE
Why use TypeScript for the Vercel templates?

### DISCUSSION THREAD

**Discussion by Elindo586** [vercel/community#4074] [Idea]
I was looking for a chatbot UI template and I noticed Vercel has something using LangChain wich is something I want to use to add to one of my Next.js  websites...

Now...  is there any specific reasons of why Vercel is creating templates with TypeScript? ...  I suppose a lot of people like TypeScript, but I prefer to use JavaScript......  Isn't easier just to add TypeScript to a template instead of deleting the TypeScript and converting it to JavaScript?

I also saw the use of TailWind, and I normally use BootStrap... that isn't all that big deal...

Anyway....  I ended up finding a little chatbot UI from W3 schools which I can modify the look, but its similar to what I want.... then I can write the logics later on to fit the Next.js platform.

**Comment by leerob** [comment]
Hey! We have some templates with TypeScript, and also some with JavaScript. Any Python too! I'd recommend using whichever toolset you feel comfortable with 😄
